tracing-appender/src/non_blocking.rs

```diff
@@ -8,14 +8,21 @@ use std::sync::Arc;
 use std::thread::JoinHandle;
 use tracing_subscriber::fmt::MakeWriter;
 
+/// Default number of lines to buffer before dropping logs if `NonBlocking` is is lossy.
```

Comment:
This constant is also used when `NonBlocking` is _not_ lossy — when the limit is reached, backpressure will be exerted on writers. How about changing this to something like
```suggestion
/// The default maximum number of buffered log lines.
///
/// If `NonBlocking` is configured to be lossy, it will drop any additional logs emitted when at
/// capacity. If it is not lossy, backpressure will be exerted on senders, causing them to wait 
/// until there is buffer capacity remaining before enqueuing new lines.
```